weave

```diff
@@ -938,6 +964,50 @@ dns_args() {
     fi
 }
 
+default_etc_hosts() {
+    NAME=$1
+    shift
+    echo "# created by Weave - BEGIN"
+    echo "# container hostname"
+    for CIDR in $ALL_CIDRS ; do
+        echo "${CIDR%/*}    $NAME"
+    done
+    echo
+    echo "# static names added with --add-host"
+    for EXTRA_HOST in "$@" ; do
+        NAME=$(echo $EXTRA_HOST | cut -f1 -d":")
+        IP=$(echo $EXTRA_HOST | cut -f2 -d":")
+        echo "$IP     $NAME"
```

Comment:
echo "${EXTRA_HOST#*:}     ${EXTRA_HOST%:*}"

(and remove the two `echo|cut` lines)